src/compile.ml

```diff
@@ -40,3 +40,14 @@ let until_interpret link_state ~unsafe ~optimize ~name m =
   let* m, link_state = until_link link_state ~unsafe ~optimize ~name m in
   let* () = Interpret.Concrete.modul link_state.envs m in
   Ok link_state
+
+let simplified_interpret link_state ~unsafe ~name m =
+  let* m =
+    if unsafe then Ok m
+    else
+      let* () = Typecheck.modul m in
+      Ok m
```

Comment:
```ocaml
let+ () = Typecheck.modul m in
m
```

:-)

Reply:
Actually, none of them should be redefined and we should simply `open Syntax`, it probably was written a long time ago. See #240 :)